Dockerfile

```diff
@@ -6,6 +6,7 @@ RUN mkdir src && touch src/lib.rs && cargo install --root /usr/local --path . ;
 ADD src ./src
 RUN touch src/lib.rs
 RUN cargo install --root /usr/local --path .
+RUN cargo install cargo-watch
```

Comment:
This `Dockerfile` is optimized for running fast when only some source code module of `check-protocols` changes. For example line 5 installs only the dependencies, so that that is cached separately from compiling the code in this repo. I would put this command at the top of the file (after `FROM ...`).